### PR TITLE
Implement get seafile link

### PR DESCRIPTION
**Pull request by Chilledheart**

### CODE REVIEW

**Comment by Chilledheart** [comment]
screen shot 2015-06-29 at 4 07 52 pm

File: src/open-local-helper.cpp [hunk 1]
```diff
@@ -94,6 +95,34 @@ void OpenLocalHelper::sendOpenLocalFileMessage(const char *url)
...
+        url.addQueryItem("email",  email);
+    url.addQueryItem("path",  path);
+#endif
+
+    return url;
+}
+
+QUrl OpenLocalHelper::generateLocalFileWebUrl(const QString& repo_id, const Account& account, const QString& path)
+{
+    return account.getAbsoluteUrl("/#common/lib/" + repo_id + (path.startsWith("/") ? path : "/" + path));
```
Comment (lins05) [inline src/open-local-helper.cpp]: The pattern `#common/lib/` is only for folders. For files we should use the pattern `https://seafile.example.com/lib/<repo-id>/file/encoded-filename.txt`

File: src/open-local-helper.cpp [hunk 2]
```diff
@@ -94,6 +95,34 @@ void OpenLocalHelper::sendOpenLocalFileMessage(const char *url)
...
+    if (!email.isEmpty())
+        url_query.addQueryItem("email",  email);
+    url_query.addQueryItem("path",  path);
+    url.setQuery(url_query);
+#else
+    url.addQueryItem("repo_id",  repo_id);
+    if (!email.isEmpty())
+        url.addQueryItem("email",  email);
+    url.addQueryItem("path",  path);
+#endif
```
Comment (lins05) [inline src/open-local-helper.cpp]: There is a convenient method [includeQueryParams](https://github.com/haiwen/seafile-client/blob/v4.2.5/src/utils/utils.h#L89) which wraps the difference of qt4 and qt5.

File: src/repo-service-helper.cpp [hunk 3]
```diff
@@ -57,7 +57,7 @@ void FileDownloadHelper::onCancel()
         disconnect(req_, 0, this, 0);
 }
 
-void FileDownloadHelper::onGetDirentsSuccess(const QList<SeafDirent> &dirents)
+void FileDownloadHelper::onGetDirentsSuccess(bool current_readonly, const QList<SeafDirent> &dirents)
```
Comment (lins05) [inline src/repo-service-helper.cpp]: the param `current_readonly` seems not used in this function?

File: src/filebrowser/seafilelink-dialog.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    hlayout->addWidget(spacer3);
+
+    QPushButton *ok = new QPushButton(tr("OK"));
+    hlayout->addWidget(ok);
+    connect(ok, SIGNAL(clicked()), this, SLOT(accept()));
+    ok->setFocus();
+
+    layout->addLayout(hlayout);
+
+    setLayout(layout);
```
Comment (lins05) [inline src/filebrowser/seafilelink-dialog.cpp]: Why not use qt designer? It's much simpler and intuitive than writing code to do the ui layouts.

File: src/open-local-helper.cpp (hunk 1 above)
Comment (Chilledheart) [inline src/open-local-helper.cpp]: exactly

File: src/open-local-helper.cpp [hunk 5]
```diff
@@ -94,6 +95,38 @@ void OpenLocalHelper::sendOpenLocalFileMessage(const char *url)
...
+    return url;
+}
+
+QUrl OpenLocalHelper::generateLocalFileWebUrl(const QString& repo_id, const Account& account, const QString& path)
+{
+    QString fixed_path = path.startsWith("/") ? path : "/" + path;
+    if (fixed_path.endsWith("/"))
+        return account.getAbsoluteUrl("/#common/lib/" + repo_id + fixed_path);
+    else
+        return account.getAbsoluteUrl("/lib/" + repo_id + "/file" + fixed_path);
```
Comment (lins05) [inline src/open-local-helper.cpp]: `fixed_path` must be url encoded.

**Pull request merged**